modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConcurrentStripedPool.java

```diff
@@ -123,37 +123,29 @@ public void clear() {
     /** {@inheritDoc} */
     @Override public @NotNull Iterator<E> iterator() {
         return new Iterator<E>() {
-            int idx = 0;
-            Iterator<E> it = stripePools[idx].iterator();
+            private int idx;
+
+            private Iterator<E> it = stripePools[idx].iterator();
 
             @Override public boolean hasNext() {
-                if (it.hasNext())
-                    return true;
+                while (true) {
+                    if (it.hasNext())
+                        return true;
 
-                idx++;
+                    if (++idx >= stripes)
+                        break;
 
-                if (idx < stripes) {
                     it = stripePools[idx].iterator();
-
-                    return it.hasNext();
                 }
-                else
-                    return false;
+
+                return false;
             }
 
             @Override public E next() {
                 if (it.hasNext())
```

Comment:
Should "it.hasNext()" be changed to "hasNext()" here?

Reply:
it MUST, thx